docs(scripts): replace deprecated prepublish and install examples with prepare

MaxBlack-dev opened the pull request:
## Description

This PR updates the scripts.md documentation to remove examples using deprecated and discouraged features.

## Changes

- **Examples section**: Replaced install/postinstall examples with prepare and 	est scripts
- **Examples section**: Replaced discouraged make commands with modern build tools (	sc, jest)
- **Use Cases section**: Changed recommendations from deprecated prepublish to prepare
- **Use Cases section**: Updated CoffeeScript example to TypeScript (more current)

## Fixes

Closes #3992

## Context

The issue reported that the documentation was confusing because it:
1. Deprecated prepublish but then recommended using it in the Use Cases section
2. Showed examples using install scripts which are discouraged in the Best Practices section
3. Used outdated examples (make, uninstall, CoffeeScript)

This update aligns the examples and recommendations with current best practices:
- Use prepare instead of the deprecated prepublish
- Avoid install scripts as recommended in Best Practices
- Use modern tooling that reflects current JavaScript/TypeScript ecosystem

Commit:
docs(scripts): replace deprecated prepublish and install examples with prepare

- Updated examples section to use prepare and test scripts instead of discouraged install/postinstall
- Changed use case recommendations from prepublish to prepare
- Replaced CoffeeScript example with TypeScript (more current)
- Replaced make commands with modern build tools (tsc, jest)
- Fixes #3992

owlstronaut commented on docs/lib/content/using-npm/scripts.md:
```diff
@@ -82,17 +82,17 @@ See <https://github.com/npm/npm/issues/10074> for a much lengthier justification
 Additionally, this means that:
 
-* You can depend on `coffee-script` as a `devDependency`, and thus your users don't need to have it installed.
+* You can depend on build tools as `devDependencies`, and thus your users don't need to have them installed.
```
missing a word?

owlstronaut commented on docs/lib/content/using-npm/scripts.md:
```diff
@@ -82,17 +82,17 @@ See <https://github.com/npm/npm/issues/10074> for a much lengthier justification
 **Use Cases**
 
-If you need to perform operations on your package before it is used, in a way that is not dependent on the operating system or architecture of the target system, use a `prepublish` script.
+If you need to perform operations on your package before it is used, in a way that is not dependent on the operating system or architecture of the target system, use a `prepare` script.
```
This sentence is also very strangely composed. Give a go at rephrasing?

MaxBlack-dev replied:
Thanks for the feedback! I've rephrased both sentences:

Line 85: Changed from "If you need to perform operations on your package before it is used, in a way that is not dependent on the operating system or architecture of the target system, use a prepare script" to "Use a prepare script to perform build tasks that are platform-independent and need to run before your package is used."

The new wording is more direct and specific.

owlstronaut commented on docs/lib/content/using-npm/scripts.md:
```diff
@@ -82,17 +82,17 @@ See <https://github.com/npm/npm/issues/10074> for a much lengthier justification
 * Fetching remote resources that your package will use.
 
-The advantage of doing these things at `prepublish` time is that they can be done once, in a single place, thus reducing complexity and variability.
+The advantage of doing these things at `prepare` time is that they can be done once, in a single place, thus reducing complexity and variability.
```
I know you didn't add this, but `doing these things` feels very 3rd grade vague. Mind rewording it while you are here?

MaxBlack-dev replied:
Thanks for the feedback! I've rephrased both sentences:

Line 92: Changed from "The advantage of doing these things at prepare time is that they can be done once, in a single place, thus reducing complexity and variability" to "Running these build tasks in the prepare script ensures they happen once, in a single place, reducing complexity and variability."

The new wording is more direct and specific.

MaxBlack-dev commented:
Thanks for the feedback! I've rephrased both sentences:

1. **Line 85**: Changed from "If you need to perform operations on your package before it is used, in a way that is not dependent on the operating system or architecture of the target system, use a `prepare` script" to "Use a `prepare` script to perform build tasks that are platform-independent and need to run before your package is used."

2. **Line 92**: Changed from "The advantage of doing these things at `prepare` time is that they can be done once, in a single place, thus reducing complexity and variability" to "Running these build tasks in the `prepare` script ensures they happen once, in a single place, reducing complexity and variability."

The new wording is more direct and specific.

MaxBlack-dev reviewed (commented):
Thanks for the feedback! I've rephrased this to be more direct: 'Use a prepare script to perform build tasks that are platform-independent and need to run before your package is used.'
